packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -706,6 +706,125 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
             reusableTestsForNonBasicTiers(SubscriptionTier.PRIVILEGED, maxPrivilegedSpendingLimit);
           });
         });
+
+        describe('@hbarlimiter-batch2 Multiple users with different tiers', () => {
+          interface AliasAccountPlan {
+            aliasAccount: AliasAccount;
+            hbarSpendingPlan: IDetailedHbarSpendingPlan;
+          }
+
+          let accountPlanObject: Record<SubscriptionTier, AliasAccountPlan[]>;
+
+          const accountPlanRequirements: Record<SubscriptionTier, number> = {
+            BASIC: 2,
+            EXTENDED: 2,
+            PRIVILEGED: 2,
+          };
+
+          const createMultipleAliasAccountsWithSpendingPlans = async (
+            accountPlanRequirements: Record<SubscriptionTier, number>,
+          ) => {
+            const accountPlanObject: Record<SubscriptionTier, AliasAccountPlan[]> = {
+              BASIC: [],
+              EXTENDED: [],
+              PRIVILEGED: [],
+            };
+
+            for (const [subscriptionTier, numOfAccounts] of Object.entries(accountPlanRequirements)) {
+              for (let i = 0; i < numOfAccounts; i++) {
+                const accountCreatedResult = await createAliasAndAssociateSpendingPlan(
+                  subscriptionTier as SubscriptionTier,
+                );
+                accountPlanObject[subscriptionTier].push(accountCreatedResult);
```

Comment:
We also want to confirm that multiple accounts can be associated with the same `HbarSpendingPlan`, contributing to a common shared budget

Reply:
@victor-yanev added coverage to confirm the above! Good catch!